### ISSUE TITLE
SonarQube startup issue

### ISSUE THREAD

**Issue by deepjyotsingh** [siamaksade/openshift-cd-demo#80]
Please advice.

2018.03.08 10:50:41 INFO  app[][o.s.a.AppFileSystem] Cleaning or creating temp directory /opt/sonarqube/temp
2018.03.08 10:50:41 INFO  app[][o.s.a.es.EsSettings] Elasticsearch listening on /127.0.0.1:9001
2018.03.08 10:50:41 INFO  app[][o.s.a.p.ProcessLauncherImpl] Launch process[[key='es', ipcIndex=1, logFilenamePrefix=es]] from [/opt/sonarqube/elasticsearch]: /opt/sonarqube/elasticsearch/bin/elasticsearch -Epath.conf=/opt/sonarqube/temp/conf/es
2018.03.08 10:50:41 INFO  app[][o.s.a.SchedulerImpl] Waiting for Elasticsearch to be up and running
2018.03.08 10:50:43 INFO  app[][o.e.p.PluginsService] no modules loaded
2018.03.08 10:50:43 INFO  app[][o.e.p.PluginsService] loaded plugin [org.elasticsearch.transport.Netty4Plugin]
**2018.03.08 10:50:47 WARN  app[][o.s.a.p.AbstractProcessMonitor] Process exited with exit value [es]: 1
2018.03.08 10:50:47 INFO  app[][o.s.a.SchedulerImpl] Process [es] is stopped
2018.03.08 10:50:47 INFO app[][o.s.a.SchedulerImpl] SonarQube is stopped**

**Comment by siamaksade** [comment]
Running out of memory?

**Comment by deepjyotsingh** [comment]
No, we are good on memory. I’m running it on a server with 12 GB memory and plenty of  CPU. Do we need to do anything special while creating the PV for sonarqube?

**Comment by deepjyotsingh** [comment]
Has anyone run into this issue with Sonarqube 6.7? This looks like an image problem to me.

**Comment by deepjyotsingh** [comment]
How can we change the ownership of data directory to 'sonar' user or run the Sonarqube services as root user.

![image](https://user-images.githubusercontent.com/37116675/37456267-f4fc88fa-2864-11e8-97ab-896dd81d408e.png)

**Comment by deepjyotsingh** [comment]
I used the below template to create the persistent volume...

![image](https://user-images.githubusercontent.com/37116675/37456770-28e40aa2-2866-11e8-878b-555e9f2b4f35.png)

**Comment by ruromero** [comment]
@deepjyotsingh can you try using this repo? It uses the images from centos repository that should be able to run with restricted scc
https://github.com/RHsyseng/docker-rhel-sonarqube/tree/master/templates

The problem you have seems to be related to the permissions in the hostPath folder that belongs to `root` and indeed the script on startup tries to run the `chown` command. This should not be the case for the `registry.centos.org/rhsyseng/sonarqube:latest` image can you confirm this is the one you have or change it otherwise?

Sorry for the late reply

**Comment by ptrckq** [comment]
I'm getting same issue, using provision.sh and doing manualy. 

Sonarqube log status:

```
  | --> Scaling sonarqube-2 to 1
-- | --
  | error: update acceptor rejected sonarqube-2: pods for rc 'cicd/sonarqube-2' took longer than 600 seconds to become available
```
And pod status is: Crash Loop Back-off.


While reading other issues I thought it was caused by lack of memory, so I increased VM Memory size to 12GB but it still not working.

I'm using Minishift as environment. Could anyone help me with this?

**Comment by ptrckq** [comment]
I was trying with Red Hat Developers cdk. 
Now i'm using https://github.com/minishift/minishift/releases and it's working perfectly.

**Comment by ruromero** [comment]
Thanks for the feedback @ptrckq

**Comment by ecwpz91** [comment]
For me persistent deploy consistently failed using OCP 3.9 on AWS.

Workaround was to remove all components and [deploy ephemeral](https://github.com/OpenShiftDemos/openshift-cd-demo/issues/80#issuecomment-373915649).

```sh
# Remove all components
oc delete service sonarqube; oc delete deploymentconfigs sonarqube; oc delete route sonarqube; oc delete imagestreams sonarqube; oc delete pvc sonarqu
be-data
```

```sh
# Deploy ephemeral
oc new-app -f https://raw.githubusercontent.com/RHsyseng/docker-rhel-sonarqube/master/templates/sonarqube-ephemeral-template.yaml --param=SONARQUBE_SERVICE_NAME=sonarqube
```

**Comment by marcelomrwin** [comment]
Thanks, I followed your workaround and it worked like a charm.
